cmake/Modules/OpmCompile.cmake

```diff
@@ -13,11 +13,13 @@ macro (opm_compile opm)
   # all public header files are together with the source. prepend our own
   # source path to the one of the dependencies so that our version of any
   # ambigious paths are used.
-  set (${opm}_INCLUDE_DIR "${PROJECT_SOURCE_DIR}")
-  set (${opm}_INCLUDE_DIRS ${${opm}_INCLUDE_DIR} ${${opm}_INCLUDE_DIRS})
+  # set (${opm}_INCLUDE_DIR "${PROJECT_SOURCE_DIR}")
+  # set (${opm}_INCLUDE_DIRS ${${opm}_INCLUDE_DIR} ${${opm}_INCLUDE_DIRS})
 
   # create this library, if there are any compilation units
-  include_directories (${${opm}_INCLUDE_DIRS})
+  # include_directories (${${opm}_INCLUDE_DIRS})
+  include_directories("${PROJECT_SOURCE_DIR}")
+  include_directories (SYSTEM ${${opm}_INCLUDE_DIRS})
```

Comment:
I guess we should make it an option? TREAT_OTHER_MODULES_AS_SYSTEM_INCLUDES is a bit long, but quite clear. However @andlaus thinks this causes a separate (?) problem (bottom of discussion in #207) and both versions of an options should be valid.